internal/indexer/indexer.go

```diff
@@ -280,36 +291,78 @@ func (i *Indexer) emitDocument(filename string) {
 	i.defined[filename] = map[int]struct{}{}
 }
 
-// addImports modifies the definitions map of each file to include entries for import statements so
+// emitImports modifies the definitions map of each file to include entries for import statements so
 // they can be indexed uniformly in subsequent steps.
-func (i *Indexer) addImports() {
-	i.visitEachPackage("Adding import definitions", i.addImportsToPackage)
+func (i *Indexer) emitImports() {
+	i.visitEachPackage("Adding import definitions", i.emitImportsForPackage)
+}
+
+// emitImportMonikerReference will emit the associated reference to the import moniker.
+// This will emit the reference in either case:
+//
+//    import "fmt"
+//            ^^^------ reference github.com/golang/go/std/fmt
+//
+//    import f "fmt"
+//              ^^^---- reference github.com/golang/go/std/fmt
+//
+// In both cases, this will emit the corresponding import moniker for "fmt". This is ImportSpec.Path
+func (i *Indexer) emitImportMonikerReference(p *packages.Package, pkg *packages.Package, spec *ast.ImportSpec) {
+	state := makePkgNameState(i, p, spec.Path.Pos(), spec.Path.Value, pkg)
+
+	i.emitImportMoniker(state.resultSetID, p, state.obj)
+
+	// @eric -- we could potentially skip this because we have am oniker now.
+	// but this breaks a test and the current structure.
```

Comment:
I think it's fine to emit the text here - we cache the hover results and we already emit it on any package reference so worrying about it _here_ doesn't fix much.

We could try to drop all cached hover references to external packages assuming we have good enough index coverage in the wild in the future.